exporters/elasticsearch-exporter/src/main/java/io/camunda/zeebe/exporter/ElasticsearchClient.java

```diff
@@ -128,32 +128,36 @@ public void flush() {
       throw new ElasticsearchExporterException("Failed to flush bulk", e);
     }
 
-    final var success = checkBulkResponse(bulkResponse);
-    if (!success) {
-      throw new ElasticsearchExporterException("Failed to flush all items of the bulk");
+    final var failure = checkBulkResponse(bulkResponse);
+    if (failure.isPresent()) {
+      throw failure.get();
     }
 
     // all records where flushed, create new bulk request, otherwise retry next time
     bulkRequest = new ArrayList<>();
   }
 
-  private boolean checkBulkResponse(final BulkResponse bulkResponse) {
+  private Optional<RuntimeException> checkBulkResponse(final BulkResponse bulkResponse) {
     final var hasErrors = bulkResponse.hasErrors();
-    if (hasErrors) {
-      bulkResponse.getItems().stream()
-          .flatMap(item -> Optional.ofNullable(item.getIndex()).stream())
-          .flatMap(index -> Optional.ofNullable(index.getError()).stream())
-          .collect(Collectors.groupingBy(BulkItemError::getType))
-          .forEach(
-              (errorType, errors) ->
-                  log.warn(
-                      "Failed to flush {} item(s) of bulk request [type: {}, reason: {}]",
-                      errors.size(),
-                      errorType,
-                      errors.get(0).getReason()));
+    if (!hasErrors) {
+      return Optional.empty();
     }
 
-    return !hasErrors;
+    final var collectedErrors = new ArrayList<String>();
+    bulkResponse.getItems().stream()
+        .flatMap(item -> Optional.ofNullable(item.getIndex()).stream())
+        .flatMap(index -> Optional.ofNullable(index.getError()).stream())
+        .collect(Collectors.groupingBy(BulkItemError::getType))
+        .forEach(
+            (errorType, errors) ->
+                collectedErrors.add(
+                    String.format(
+                        "Failed to flush %d item(s) of bulk request [type: %s, reason: %s]",
+                        errors.size(), errorType, errors.get(0).getReason())));
```

Comment:
Why we only append the first error reason? Are they always the same?

Reply:
:man_facepalming: 

I just kept the previous structure without thinking, but you're right, we should just throw :sweat_smile: Sorry, I wasn't thinking

Reply:
I assume - it was already like this, so I would assume whoever initially wrote this also double checked that. Do we think the reason may sometimes be different? Would it matter much?